docs/FAQ.md

```diff
@@ -0,0 +1,236 @@
+  Int:
+    type: int32
+```
+
+Or, you can bind it to any other type, perhaps one with size-checked constructors; see the [`genqlient.yaml` documentation](`genqlient.yaml`) for more details.
+
+## How do I make a query with …
+
+### … a specific name for a field?
+
+genqlient supports GraphQL field-aliases, and uses them to determine the Go struct field name.  For example, if you do
+```graphql
+query MyQuery {
+  myGreatName: myField
+}
+```
+and genqlient will generate a Go field `MyGreatName string`.  Note that the alias will always be uppercased, to ensure the field is visible to the Go JSON library.
+
+### … nullable fields?
+
+There are two ways to handle nullable fields in genqlient.  One way is to use the Go idiom, where null gets mapped to the zero value; this is the default in genqlient.  So if you have a GraphQL field of type `String`, and you do:
+
+```graphql
+query MyQuery(arg: String) {
+  myField
+}
+```
+
+then genqlient will generate a Go field `MyField string`, and set it to the empty string if the server returns null.  This works even for structs: if an object type in GraphQL is null, genqlient will set the corresponding struct to its zero value.  It can be helpful to request `id` in such cases, since that’s a field that should always be set, or `__typename` which is guaranteed to be set, so you can use its presence to decide whether to look at the other fields.
+
+For input fields, you often want to tell genqlient to send null to the server if the argument is set to the zero value, similar to the JSON `omitempty` tag.  In this case, you can do:
+
+```graphql
+query MyQuery(
+  # @genqlient(omitempty: true)
+  arg: String,
+) {
+  myField
+}
+```
+
+You can also put the `# @genqlient(omitempty: true)` on the first line, which will apply it to all arguments in the query.
+
+If you need to distinguish null from the empty string (or generally from the Go zero value of your type), you can tell genqlient to use a pointer for the field or argument like this:
+```graphql
+query MyQuery(
+  # @genqlient(pointer: true)
+  arg: String,
+) {
+  # @genqlient(pointer: true)
+  myField
+}
+```
+
+This will generate a Go field `MyField *string`, and set it to `nil` if the server returns null (and in reverse for arguments).  Such fields can be harder to work with in Go, but allow a clear distinction between null and the Go zero value.
+
+See [genqlient_directive.graphql](genqlient_directive.graphql) for complete documentation on these options.
+
+### … GraphQL interfaces?
+
+If you request an interface field, genqlient generates an interface type corresponding to the GraphQL interface, and several struct types corresponding to its implementations.  For example, given a query:
+
+```graphql
+query GetBooks {
+  favorite {
+    title
+    ... on Novel {
+      protagonist
+    }
+    ... on Dictionary {
+      language
+    }
+  }
+}
+```
+
+genqlient will generate the following types:
+
+```go
+type GetBooksFavoriteBook interface {
```

Comment:
This might be documented elsewhere, but how does genqlient generate the `FavouriteBook` part of the name?

Reply:
Yep it's below!  I can mention it.